### PR TITLE
Replace old RFC1123 impl which cannot be built until GCC 5

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**
https://github.com/Azure/azure-storage-cpplite/commit/f1444b4ceec8ca44332146df6a919997deb998c5#commitcomment-41366200

### CODE REVIEW

File: src/utility.cpp [hunk 1]
```diff
@@ -98,10 +96,14 @@ namespace azure {  namespace storage_lite {
         }
         else if (format == date_format::rfc_1123)
         {
-            std::stringstream ss;
-            ss.imbue(std::locale("C"));
-            ss << std::put_time(pm, constants::date_format_rfc_1123);
-            return ss.str();
+            const char* weekdays[] = { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat" };
```
Comment (katmsft) [inline src/utility.cpp:99]: If the compiler does not optimize this, use static keyword to avoid recreating the arrays please.

Comment (JinmingHu-MSFT) [reply src/utility.cpp]: fixed

**Comment by npritchard** [comment]
Thanks, team!

**Pull request merged**